### PR TITLE
Update Resource Naming Restrictions

### PR DESCRIPTION
**Pull request by bennerv**
Updates resource naming restrictions for `hcpOpenShiftClusters` and `hcpOpenShiftClusters/nodePools`

### CODE REVIEW

File: frontend/pkg/frontend/const.go [hunk 1]
```diff
@@ -17,4 +17,7 @@ const (
 	PathSegmentDeploymentName    = "deploymentname"
 	PathSegmentActionName        = "actionname"
 	PathSegmentNodepoolName      = "nodepoolname"
+
+	HCPClusterResourceType = "hcpopenshiftclusters"
+	NodePoolResourceType   = "hcpopenshiftclusters/nodepools"
```
Comment (mbarnes) [inline frontend/pkg/frontend/const.go:22]: Can you move the node pool constant to `internal/api/registry.go`?
We already have a constant for cluster resource type defined there.

**Review by mbarnes — DISMISSED** [review]
Just a small nit about where constants are defined.

File: frontend/pkg/frontend/const.go (hunk 1 above)
Comment (bennerv) [reply frontend/pkg/frontend/const.go:22]: Done

**Review by mbarnes — DISMISSED** [review]

**Comment by bennerv** [comment]
There's going to be a problem with nesting node pool resource types as the resource type name.  Need to fix it quick.  Moving to draft.

**Review by mjlshen — APPROVED** [review]

**Pull request merged**